app/controllers/blogs_controller.rb

```diff
@@ -3,13 +3,17 @@
 class BlogsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index show]
 
-  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :set_blog, only: %i[edit update destroy]
 
   def index
     @blogs = Blog.search(params[:term]).published.default_order
   end
 
-  def show; end
+  def show
+    @blog = Blog.find(params[:id])
+
+    raise ActiveRecord::RecordNotFound if @blog.secret && @blog.user != current_user
```

Comment:
この書き方でもテストはパスしますが、理想を言うと、

```suggestion
    @blog = Blog.???.find(params[:id])
```

↑こんな感じで、検索条件(scope)に合致しなければ自動的にRecordNotFoundが発生するようにするのがベストです

Reply:
https://github.com/outa1127/haunted-blog/pull/1/commits/359a74fe47010289f2c8d94504d2fc11fd7c4278 にてモデルにscopeを追加し、詳細ページを表示する際にsecretフラグがfalseのものかログインしているユーザーが追加したものは表示できるように修正いたしました。